File: django_backend/project/urls.py
from django.urls import include, path
from django.contrib import admin
from rest_framework import routers

#from user_profile.views import UserViewSet, EmployeeViewSet, UserProfileViewSet
from users.views import CustomUserViewSet, EmployeeViewSet, MyEmployeeViewSet, Employers
from videoinwerken_employee.models import EmployeeQuestion
from videoinwerken_employee.views import EmployeeCourseView, EmployeeQuestionView, EmployeeCourseQuestionsView, \
    EmployeeQuestionsCourse
from videoinwerken_manager.views import CourseView, CourseViewMyCourses, CourseQuestionView, CourseQuestionAnswerView

router = routers.DefaultRouter()
#router.register('accounts', UserViewSet)
#router.register('accountsprofile', UserProfileViewSet)
router.register('customusers', CustomUserViewSet)
router.register('manager/myemployees', MyEmployeeViewSet)
router.register('manager/employees', EmployeeViewSet)

router.register('employers', Employers)

router.register('manager/mycourses', CourseViewMyCourses) # list courses of manager
router.register('manager/courses', CourseView) # add/update/delete courses
router.register('manager/course/questions', CourseQuestionView)
router.register('manager/course/question/answers', CourseQuestionAnswerView)

# employee-questions pairs for course_id and employee
router.register('employee/employeequestionscourse', EmployeeQuestionsCourse)

#employees
router.register('employee/courses', EmployeeCourseView)

# Get course-questions pairs for an employee
router.register('employee/coursequestions', EmployeeCourseQuestionsView)

# Employee question record, use to see if attempted/passed the question
router.register('employee/employeequestion', EmployeeQuestionView)

urlpatterns = [
    path('admin/', admin.site.urls),

    # This is used for user reset password
    path('', include('django.contrib.auth.urls')),
    path('rest-auth/', include('rest_auth.urls')),
    path('rest-auth/registration/', include('rest_auth.registration.urls')),
    path('account/', include('allauth.urls')),

    path('api/',  include(router.urls)),
]

File: react_frontend/src/constants/urls.js
//const ROOT_URL = "http://localhost:8000/";

//const ROOT_URL = "http://h2900735.stratoserver.net:8000/";
const ROOT_URL = "http://0.0.0.0:8000/";

export const AuthUrls = {
  LOGIN: `${ROOT_URL}rest-auth/login/`,
  SIGNUP: `${ROOT_URL}rest-auth/registration/`,
  CHANGE_PASSWORD: `${ROOT_URL}rest-auth/password/change/`,
  RESET_PASSWORD: `${ROOT_URL}rest-auth/password/reset/`,
  RESET_PASSWORD_CONFIRM: `${ROOT_URL}rest-auth/password/reset/confirm/`,
  USER_ACTIVATION: `${ROOT_URL}rest-auth/registration/verify-email/`,
  USER_PROFILE: `${ROOT_URL}rest-auth/user/`,

  COURSES: `${ROOT_URL}api/manager/courses/`,

  API_USER: `${ROOT_URL}api/getuser/`,
  API_EMPLOYERS: `${ROOT_URL}api/employers/`,

  API_EMPLOYEE_COURSES: `${ROOT_URL}api/employee/courses/`,
  API_EMPLOYEE_EMPLOYEEQUESTIONSCOURSE: `${ROOT_URL}api/employee/employeequestionscourse/`,
  API_EMPLOYEE_COURSEQUESTIONS: `${ROOT_URL}api/employee/coursequestions/`,
  API_EMPLOYEE_EMPLOYEEQUESTION: `${ROOT_URL}api/employee/employeequestion/`,

  API_MANAGER_COURSES: `${ROOT_URL}api/manager/courses/`,
  API_MANAGER_COURSE_QUESTIONS: `${ROOT_URL}api/manager/course/questions/`,
  API_MANAGER_COURSE_QUESTION_ANSWERS: `${ROOT_URL}api/manager/course/question/answers/`,
};
